api/v1beta1/types.go

```diff
@@ -31,6 +31,28 @@ const (
 	Bastion string = "bastion"
 )
 
+// SecurityEncryptionType represents the Encryption Type when the virtual machine is a
+// Confidential VM.
+type SecurityEncryptionType string
```

Comment:
This type and constants redefine some from [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/resourcemanager/compute/armcompute/constants.go#L1722). Could we use those directly instead?

Reply:
There are good reasons for not using types directly from the Azure Go SDK. We don't want to have a dependency on a particular SDK version in our API as it might result in breaking changes between API versions.

Redefining some of the SDK types and constants is a side effect of that, and what we've done for all other existing types. 
 
+1 for keeping it as-is

Reply:
@CecileRobertMichon thank you for the clarification, it's helpful as to where we should be steering towards. Looking at the codebase, I think there are fields defined as generic `string`s, whereas they are in fact `enum`s (e.g. the [one](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go#L616) mentioned above). Those get validated via the respective `azure-sdk-for-go` `enum`s. If I understand correctly, these validations will take place irrespective of whether we would define those `enum`s in the API or just leave them as generic `string`s. 

But IMHO, having the `enum`s defined here (even if they seem as duplicates of `azure-sdk-for-go`), communicates clearly the separation of our API from `azure-sdk-for-go`. Which potentially helps users easily identify what the current version of the API supports, regarding those particular types, irrespective of `azure-sdk-for-go`.

I'm happy to go with either implementation, as long as it adheres to where we should be heading (but I do have to admit that I would go with keeping it as-is as well :) ).

Reply:
Makes sense to me: let's leave it as is.